src/ray/common/ray_syncer/ray_syncer.cc

```diff
@@ -382,80 +383,25 @@ void RaySyncer::BroadcastMessage(std::shared_ptr<const RaySyncMessage> message)
-    grpc::CallbackServerContext *context,
-    const StartSyncRequest *request,
-    StartSyncResponse *response) {
-  auto *reactor = context->DefaultReactor();
-  // Make sure server only have one client
-  if (!remote_node_id_.empty()) {
-    RAY_LOG(WARNING) << "Get a new sync request from "
-                     << NodeID::FromBinary(request->node_id()) << ". "
-                     << "Now disconnect from " << NodeID::FromBinary(remote_node_id_);
-    syncer_.Disconnect(remote_node_id_);
-  }
-  remote_node_id_ = request->node_id();
-  RAY_LOG(DEBUG) << "Get connect from: " << NodeID::FromBinary(remote_node_id_);
-  syncer_.GetIOContext().dispatch(
-      [this, response, reactor, context]() {
-        if (context->IsCancelled()) {
-          reactor->Finish(grpc::Status::CANCELLED);
-          return;
-        }
-
-        syncer_.Connect(std::make_unique<ServerSyncConnection>(
-            syncer_.GetIOContext(), remote_node_id_, [this](auto msg) {
-              syncer_.BroadcastMessage(msg);
-            }));
-        response->set_node_id(syncer_.GetLocalNodeID());
-        reactor->Finish(grpc::Status::OK);
-      },
-      "RaySyncer::StartSync");
-  return reactor;
-}
-
-grpc::ServerUnaryReactor *RaySyncerService::Update(grpc::CallbackServerContext *context,
-                                                   const RaySyncMessages *request,
-                                                   DummyResponse *) {
-  auto *reactor = context->DefaultReactor();
-  // Make sure request is allocated from heap so that it can be moved safely.
-  RAY_CHECK(request->GetArena() == nullptr);
-  syncer_.GetIOContext().dispatch(
-      [this, request = std::move(*const_cast<RaySyncMessages *>(request))]() mutable {
-        auto *sync_connection = dynamic_cast<ServerSyncConnection *>(
-            syncer_.GetSyncConnection(remote_node_id_));
-        if (sync_connection != nullptr) {
-          sync_connection->ReceiveUpdate(std::move(request));
-        } else {
-          RAY_LOG(FATAL) << "Fail to get the sync context";
-        }
-      },
-      "SyncerUpdate");
-  reactor->Finish(grpc::Status::OK);
-  return reactor;
-}
-
-grpc::ServerUnaryReactor *RaySyncerService::LongPolling(
-    grpc::CallbackServerContext *context,
-    const DummyRequest *,
-    RaySyncMessages *response) {
-  auto *reactor = context->DefaultReactor();
-  syncer_.GetIOContext().dispatch(
-      [this, reactor, response]() mutable {
-        auto *sync_connection = dynamic_cast<ServerSyncConnection *>(
-            syncer_.GetSyncConnection(remote_node_id_));
-        if (sync_connection != nullptr) {
-          sync_connection->HandleLongPollingRequest(reactor, response);
-        } else {
-          RAY_LOG(ERROR) << "Fail to setup long-polling";
-          reactor->Finish(grpc::Status::CANCELLED);
-        }
-      },
-      "SyncLongPolling");
-  return reactor;
+ServerBidiReactor *RaySyncerService::StartSync(grpc::CallbackServerContext *context) {
```

Comment:
should we return an unique_ptr since we release the ownership as well?

Reply:
This part belong to gRPC, so we can't change this.